Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs

```diff
@@ -225,5 +240,93 @@ private void ExportToCSV(object sender, System.Windows.RoutedEventArgs e)
             Export.GingerToCSV.BrowseForFilename();
             Export.GingerToCSV.BusinessFlowToCSV(mBusinessFlow);
         }
+
+        private IEnumerable<ActivitiesGroup> GetActivityGroupsMissingFromSharedRepository()
+        {
+            return mBusinessFlow.ActivitiesGroups.Where(ag => !ag.IsSharedRepositoryInstance);
+        }
+
+        private void ExportBPMNMenuItem_Click(object sender, RoutedEventArgs e)
+        {
+            bool wasAllAddedToSharedRepository = TryAddingMissingActivityGroupsToSharedRepository();
+            if (!wasAllAddedToSharedRepository)
+            {
+                return;
+            }
+            ExportBusinessFlowBPMN();
+        }
+
+        private bool TryAddingMissingActivityGroupsToSharedRepository()
+        {
+            bool wasAllAddedToSharedRepository; 
+            try
+            { 
+                IEnumerable<ActivitiesGroup> activityGroups = GetActivityGroupsMissingFromSharedRepository();
+                bool hasActivityGroupsMissingFromSharedRepository = activityGroups.Any();
+                if (hasActivityGroupsMissingFromSharedRepository)
+                {
+                    wasAllAddedToSharedRepository = true;
+                    return wasAllAddedToSharedRepository;
+                }
+
+                eUserMsgSelection userResponse = Reporter.ToUser(eUserMsgKey.AddActivityGroupsToSharedRepositoryForBPMNConversion);
+                if (userResponse != eUserMsgSelection.Yes)
+                {
+                    wasAllAddedToSharedRepository = false;
+                    return wasAllAddedToSharedRepository;
+                }
+
+                Context context = new()
+                {
+                    BusinessFlow = mBusinessFlow
+                };
+                IEnumerable<RepositoryItemBase> activitiesAndGroups = activityGroups
+                    .SelectMany(ag => ag.ActivitiesIdentifiers.Select(ai => ai.IdentifiedActivity))
+                    .Cast<RepositoryItemBase>()
+                    .Concat(activityGroups);
+                WizardWindow.ShowWizard(new UploadItemToRepositoryWizard(context, activitiesAndGroups));
+
+                wasAllAddedToSharedRepository = activityGroups.All(ag => ag.IsSharedRepositoryInstance);
+                return wasAllAddedToSharedRepository;
+            }
+            catch(Exception ex)
+            {
+                Reporter.ToUser(eUserMsgKey.FailedToAddItemsToSharedRepository, "Unexpected error, check logs for more details");
+                Reporter.ToLog(eLogLevel.ERROR, "Error occurred while adding missing activities and activity groups to shared repository.", ex);
+                wasAllAddedToSharedRepository = false;
+                return wasAllAddedToSharedRepository;
```

Comment:
Add return statement to Finally as we same flag is being returned from both try and catch block

Reply:
As discussed, we won't do this modification